addons/web/static/src/views/relational_model/static_list.js

```diff
@@ -681,10 +695,11 @@ export class StaticList extends DataPoint {
             return compareRecords(r1, r2, orderBy, this.fields) || (orderBy[0].asc ? -1 : 1);
         });
         const currentPageRecords = sortedRecords.slice(this.offset, this.offset + this.limit);
-        //TODO: read records that haven't been fully read yet
         this.model._updateConfig(this.config, { orderBy }, { noReload: true });
+        this._currentIds = sortedRecords.map((r) => r.resId || r.virtualId);
         this.records = currentPageRecords;
         this._needsReordering = false;
+        await this._load();
```

Comment:
I don't think it's the best way to go, as you update this.records before records are loaded. What about calling _loadRecords (cf line 686) with the activeFields of the config, for all not fully loaded yet records?